### DISCUSSION TITLE
New config system (eslint.config.js): how to import globals browser

### DISCUSSION THREAD

**Discussion by LaFrimousse** [eslint/eslint#17152] [Q&A] [answered]
Hi,

I would like to to import the **globals: { ...globals.browser }** in my eslint.config.js config file.
(My motivation is to get rid of the eslint warning ```'console' is not defined ``` I get when using the rule ```"no-unused-vars": "warn"```. )

I took example of the config proposed in the [ESLint's new config system, Part 2](https://eslint.org/blog/2022/08/new-config-system-part-2/#goodbye-environments%2C-hello-globals) and the following is my actual eslint.config.js file:

```
import typescriptParser from "@typescript-eslint/parser";
import react from "eslint-plugin-react";
import globals from "globals";

export default [
	{
		files: ["**/*.{js,jsx,mjs,cjs,ts,tsx}"],
		languageOptions: {
			ecmaVersion: "latest",
			sourceType: "module",
			globals: {
				...globals.browser,
			},
			parser: typescriptParser,
		},
		plugins: {
			react: react,
		},
		rules: {
			"no-unused-vars": "warn",
		},
	},
];
```
But when I try to run eslint: 
```$ npx eslint src/pages/Tab1.tsx```

I get the following error:
```
Oops! Something went wrong! :(

ESLint: 8.39.0

TypeError: Key "languageOptions": Key "globals": Global "AudioWorkletGlobalScope " has leading or trailing whitespace.
    at Object.validate (/path-to-my-project/node_modules/eslint/lib/config/flat-config-schema.js:219:23)
    at ObjectSchema.validate (/path-to-my-project/node_modules/@humanwhocodes/object-schema/src/object-schema.js:218:35)
    at Object.validate (/path-to-my-project/node_modules/@humanwhocodes/object-schema/src/object-schema.js:112:32)
    at ObjectSchema.validate (/path-to-my-project/node_modules/@humanwhocodes/object-schema/src/object-schema.js:218:35)
    at /path-to-my-project/node_modules/@humanwhocodes/object-schema/src/object-schema.js:171:18
    at Array.reduce (<anonymous>)
    at ObjectSchema.merge (/path-to-my-project/node_modules/@humanwhocodes/object-schema/src/object-schema.js:169:24)
    at /path-to-my-project/node_modules/@humanwhocodes/config-array/api.js:884:42
    at Array.reduce (<anonymous>)
    at FlatConfigArray.getConfig (/path-to-my-project/node_modules/@humanwhocodes/config-array/api.js:883:39)
```

Can somebody please explain me how to deal with that problem ? 

Thanks!

**Comment by aladdin-add** [comment, marked as answer]
Which version of `globals` are you using?  It's an issue appeared in some previous version, but it should have been fixed. you can upgrade the package.

related: https://github.com/sindresorhus/globals/pull/184

**Reply by LaFrimousse** [reply]
You got this one right ! Thanks a lot !

I did not explicitly installed 'globals', it was a secondary dependency resolved to 11.12.0 (from 2019).
Even though eslint was requesting globals "^13.19.0", @babel with dependency to 'globals' "^11.1.0"  was winning the competition.


`$ npm install globals` solved the problem :)

**Reply by ryancwalsh** [reply]
This helped me too. Thanks!
